### PR TITLE
Refresh astro.new

### PR DESCRIPTION
**Pull request by jasikpark**
This PR:
- sets up `astro/tsconfigs/strictest`
- fixes the discovered errors
- sets up prettier, prettier-plugin-astro, prettier-plugin-tailwindcss
- formats accordingly
- Updates to astro 1.3.0

This was kinda a mess since I did formatting and types in the same PR 😅 

Read the commits, one by one

### CODE REVIEW

File: tsconfig.json
Comment (Princesseuh) [inline tsconfig.json:14]: You don't need all of this with the extends

File: .github/workflows/lint.yml
Comment (delucis) [inline .github/workflows/lint.yml:18]: In astro & docs we don't fail CI for prettier, we just have an action that formats and auto commits the result if needs be. Don't know what you think but it's slightly friendlier.

**Review by delucis — COMMENTED** [review]
Seems good to me! Left one comment for your consideration.

File: .github/workflows/lint.yml
Comment (jasikpark) [reply .github/workflows/lint.yml:18]: do we have a reusable workflow / github action for that? what perms does it need?

Comment (jasikpark) [reply .github/workflows/lint.yml]: Okay, pushed the format action, though I don't know whether it has perms; copied it from prettier-plugin-astro

File: .github/workflows/format.yml
Comment (delucis) [inline .github/workflows/format.yml:26]: Not sure how crucial this is, but elsewhere we specify FredKBot as the committer:

```suggestion
          branch: ${{ github.head_ref }}
          commit_user_name: fredkbot
          commit_user_email: fred+astrobot@astro.build
```

We may need to add the token to the repo secrets. @FredKSchott is probably the only one with permissions to do that.

Comment (jasikpark) [reply .github/workflows/format.yml:26]: hmmm okay; not the case in prettier-plugin-astro ig

Comment (Princesseuh) [reply .github/workflows/format.yml:26]: This only really matters on the main repo, where the main branch is protected

Comment (jasikpark) [reply .github/workflows/format.yml:26]: so what's the verdict? I can probably just leave it as is, right?

Comment (delucis) [reply .github/workflows/format.yml:26]: This is a conspiracy to keep FredKBot’s GitHub stats down!!! 😡

**Review by delucis — APPROVED** [review]
LGTM! Let’s see if we do need to add that token…

**Comment by jasikpark** [comment]
ack... do i need to turn on "pls don't look at my deps"?

**Comment by delucis** [comment]
LOL, classic “just one more small thing” effect 😆

Yeah, enabling `skipLibCheck` should fix this error.

**Review by delucis — APPROVED** [review]
Ship it!

**Pull request merged**
